src/Common/ExponentiallySmoothedCounter.h

```diff
@@ -485,4 +485,457 @@ struct ExponentiallySmoothedAlphaWithTimeFillGaps
     }
 };
 
+struct Holt
+{
+    double value = 0;
+
+    double trend = 0;
+
+    unsigned long long int count = 0;
+
+    double first_value;
+
+    double first_trend;
+
+    Holt() = default;
+
+    Holt(double current_value)
+        : value(current_value), trend(0), count(1), first_value(current_value), first_trend(0)
+    {
+    }
+
+    Holt(double current_value, double current_trend, unsigned long long int current_count, double first_value_, double first_trend_)
+        : value(current_value), trend(current_trend), count(current_count), first_value(first_value_), first_trend(first_trend_)
+    {
+    }
+
+    static double scale(unsigned long long int count_passed, double value)
+    {
+        /// Using binary power because of low precision of pow().
+        double result = 1;
+        value = 1 - value;
+        while (count_passed)
+        {
+            if (count_passed & 1)
+            {
+                result *= value;
+            }
+            count_passed >>= 1;
+            value *= value;
+        }
+        return result;
+    }
+
+    Holt remap(unsigned long long int current_count, double alpha, double beta) const
+    {
+        // approximation formula for remapping
```

Comment:
Why is it not possible to calculate an exact remapped value given alpha and beta?